pubtools/pulplib/_impl/client/client.py

```diff
@@ -239,10 +284,18 @@ def _unpack_response(cls, pulp_response):
             parsed = pulp_response.json()
         except Exception:
             # Couldn't parse as JSON?
-            # If the response was unsuccessful, raise that.
-            # Otherwise re-raise parse error.
-            pulp_response.raise_for_status()
-            raise
+            if (
+                pulp_response.url.endswith(retry.MAINTENANCE_REPROT_URL)
```

Comment:
```suggestion
                pulp_response.url.endswith(retry.MAINTENANCE_REPORT_URL)
```

The solution here is a bit ugly though, mixing a condition specific to maintenance mode in with this code applied on all responses.

Did you also try the suggestion of *never* retrying on 404?  Then I would expect do_get_maintenance should catch the error and return None, rather than adding this special casing here.